src/overlays/actors/ovl_En_Niw/z_en_niw.c

```diff
@@ -21,61 +48,1131 @@ const ActorInit En_Niw_InitVars = {
+        if (this->actor.params == 4) {
+            this->actor.params = 0;
+        }
+        this->actionFunc = EnNiw_ResetAction;
+    } else {
+        this->unk_2E4 = Math_atan2f(this->actor.posRot.pos.x - player->actor.posRot.pos.x,
+                                    this->actor.posRot.pos.z - player->actor.posRot.pos.z) *
+                        (0x8000 / M_PI);
+        func_80AB6100(this, globalCtx, 0);
+        func_80AB5BF8(this, globalCtx, 2);
+    }
+}
+
+void func_80AB7420(EnNiw* this, GlobalContext* globalCtx) {
+    if (this->actor.bgCheckFlags & 1) {
+        this->unk_2A4 = (s16)Math_Rand_ZeroFloat(3.99f) + 5;
+        this->actionFunc = EnNiw_ResetAction;
+    }
+}
+
+void func_80AB747C(EnNiw* this, GlobalContext* globalCtx) {
+    if (this->unk_2A8 == 0 && this->actor.params != 0xA && this->actionFunc != func_80AB6450 &&
+        this->collider.base.acFlags & 2) {
+        this->collider.base.acFlags &= ~2;
+        this->sfxTimer1 = 30;
+        if (this->unk_2A4 > 0 && D_80AB85E0 == 0) {
+            this->unk_2A4--;
+        }
+        this->unk_2A6 = 1;
+        Audio_PlayActorSound2(&this->actor, NA_SE_EV_CHICKEN_CRY_M);
+        this->timer6 = 100;
+        this->path = 0;
+        this->actionFunc = func_80AB7290;
+    }
+}
+
+void EnNiw_Update(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad1;
+    EnNiw* this = THIS;
+    Player* player = PLAYER;
+    s16 i;
+    s16 featherCount;
+    Vec3f zeroVec1 = { 0.0f, 0.0f, 0.0f };
+    Vec3f zeroVec2 = { 0.0f, 0.0f, 0.0f };
+    Vec3f pos;
+    Vec3f vel;
+    Vec3f accel;
+    s32 pad2;
+    f32 scale;
+    Vec3f cam;
+    f32 dist;
+    f32 camResult;
+    s32 pad3[10];
+
+    this->unk_294++;
+
+    if (this->actionFunc != func_80AB6570) {
+        this->unk_26C[9] = 0.0f;
+    }
+    if (this->unk_2A6) {
+        featherCount = 20;
+
+        if (this->unk_2A6 == 2) {
+            featherCount = 4;
+        }
+
+        for (i = 0; i < featherCount; i++) {
+            pos.x = Math_Rand_CenteredFloat(10.0f) + this->actor.posRot.pos.x;
+            pos.y = Math_Rand_CenteredFloat(10.0f) + (this->actor.posRot.pos.y + this->unk_304);
+            pos.z = Math_Rand_CenteredFloat(10.0f) + this->actor.posRot.pos.z;
+            scale = Math_Rand_ZeroFloat(6.0f) + 6.0f;
+
+            if (this->unk_2A6 == 2 && this->unk_304 != 0) {
+                pos.y += 10;
+            }
+            if (this->unk_304 == 0) {
+                scale = Math_Rand_ZeroFloat(2.0f) + 2;
+            }
+
+            vel.x = Math_Rand_CenteredFloat(3.0f);
+            vel.y = (Math_Rand_ZeroFloat(2.0f) * 0.5f) + 2.0f;
+            vel.z = Math_Rand_CenteredFloat(3.0f);
+            accel.x = 0.0f;
+            accel.y = -0.15f;
+            accel.z = 0.0f;
+            EnNiw_FeatherSpawn(this, &pos, &vel, &accel, scale);
+        }
+
+        this->unk_2A6 = 0;
+    }
+
+    EnNiw_FeatherUpdate(&this->actor, globalCtx);
+    DECR(this->timer1);
```

Comment:
We generally avoid using DECR unless necessary because it can't be consistent since this kind of check doesn't always use `!= 0`

Reply:
DECR is actually required to match.

Reply:
are you sure? whats the diff
feel free to post in discord if its easier than here

Reply:
I honestly don't know because for some reason diff.py shows no difference, yet it does not OK